app/controllers/redirections_controller.rb

```diff
@@ -23,8 +23,17 @@ def find_or_create_redirection
   end
 
   def log_headers_and_use_fallback_redirection
-    logger.info "Redirection creation failed. Here are the headers:"
-    logger.info http_headers
+    tagged_log "Redirection creation failed."
+    if request.env["HTTP_UPGRADE_INSECURE_REQUESTS"].present?
+      tagged_log "It likely failed because the links use HTTP instead of HTTPS"
```

Comment:
This is what happened here: https://my.papertrailapp.com/events?focus=1309069283861745671&selected=1309069283861745671

Now it's easy to see that this was the cause.